server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -0,0 +1,494 @@
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("failed"))
+            .body("taskId", is(taskId))
+            .body("type", is(DeletedMessageVaultRestoreTask.TYPE))
+            .body("additionalInformation.successfulRestoreCount", is(0))
+            .body("additionalInformation.errorRestoreCount", is(0))
+            .body("additionalInformation.failRemovingCount", is(0))
+            .body("additionalInformation.user", is(USER.asString()))
+            .body("startedDate", is(notNullValue()))
+            .body("submitDate", is(notNullValue()));
+    }
+
+    @Test
+    void restoreShouldProduceFailedTaskWithFailsRemovingCountWhenTheVaultDeleteGetsError() {
+        vault.append(USER, DELETED_MESSAGE).block();
+        vault.append(USER, DELETED_MESSAGE_2).block();
+
+        doThrow(new RuntimeException("mock exception"))
+            .when(vault)
+            .delete(any(), any());
+
+        String taskId = with()
+            .post(USER.asString())
+            .jsonPath()
+            .get("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("failed"))
+            .body("taskId", is(taskId))
+            .body("type", is(DeletedMessageVaultRestoreTask.TYPE))
+            .body("additionalInformation.successfulRestoreCount", is(2))
+            .body("additionalInformation.errorRestoreCount", is(0))
+            .body("additionalInformation.failRemovingCount", is(2))
+            .body("additionalInformation.user", is(USER.asString()))
+            .body("startedDate", is(notNullValue()))
+            .body("submitDate", is(notNullValue()));
+
+        assertThat(Flux.from(vault.search(USER, Query.ALL)).toStream())
+            .containsOnly(DELETED_MESSAGE, DELETED_MESSAGE_2);
+    }
+
+    @Test
+    void restoreShouldRestoreEvenWhenVaultDeleteGetsError() throws Exception {
+        vault.append(USER, DELETED_MESSAGE).block();
+        vault.append(USER, DELETED_MESSAGE_2).block();
+
+        doThrow(new RuntimeException("mock exception"))
+            .when(vault)
+            .delete(any(), any());
+
+        String taskId = with()
+            .post(USER.asString())
+            .jsonPath()
+            .get("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("failed"));
+
+        assertThat(restoreMailboxMessages(USER))
+            .hasSize(2)
+            .anySatisfy(messageResult -> assertThat(fullContent(messageResult)).hasSameContentAs(DELETED_MESSAGE.getContent().get()))
+            .anySatisfy(messageResult -> assertThat(fullContent(messageResult)).hasSameContentAs(DELETED_MESSAGE_2.getContent().get()));
+    }
+
+    @Test
+    void restoreShouldProduceFailedTaskWithErrorRestoreCountWhenMessageAppendGetsError() throws Exception {
```

Comment:
you assert too many things in your tests. Your assertions should reflect the intent of the test (~ the test name) and not more.
The more you write assertions, the more you make it hard the change the software.